### ISSUE TITLE
p910nd: Process p9100d disappears after some time

### ISSUE THREAD

**Issue by triebelf** [openwrt/packages#7831]
Maintainer: @pkerling 
Environment: ARMv7 Processor rev 1 (v7l), Linksys WRT3200ACM, OpenWrt 18.06.1

Description:
After some time (don't know when), process `p9100d` just goes away. Logread or `ps | grep p9100d` shows nothing. The printer is power on very infrequently. After doing `/etc/init.d/p910nd restart`, everything works fine for a while. How can I diagnose and fix this?

I also see this, service is enabled:
```
> ls -l /etc/rc.d/[SK]??p910nd
lrwxrwxrwx    1 root     root            16 Nov 30 19:43 /etc/rc.d/S50p910nd -> ../init.d/p910nd
```

**Comment by diizzyy** [comment]
Does logread say anything?

**Comment by cshoredaniel** [comment]
Is the device p9100d is supposed to be sharing effectively unplugging and then reconnecting?  Will need to check the hotplug (or lack thereof) script as I'm looking as using this myself soon.

**Comment by triebelf** [comment]
Logread output when powering printer on/off:
```
Wed Jan  2 20:08:02 2019 kern.info kernel: [627000.780875] usb 1-1: new full-speed USB device number 5 using orion-ehci
Wed Jan  2 20:08:02 2019 kern.info kernel: [627000.991528] usblp 1-1:1.0: usblp0: USB Bidirectional printer dev 5 if 0 alt 0 proto 2 vid 0x04E8 pid 0x326C
Wed Jan  2 20:08:11 2019 kern.info kernel: [627009.446621] usb 1-1: USB disconnect, device number 5
Wed Jan  2 20:08:11 2019 kern.info kernel: [627009.451809] usblp0: removed
```
Does this mean hotplug? Tell me more about a missing script.

**Comment by cshoredaniel** [comment]
@triebelf yep, that means the device in /dev goes away (or at least the app can't use it), so what's needed is probably a a mod to /etc/hotplug.json that does the same thing as the tty subsystem.
The tty subsystem does this:
```
        [ "if",
                [ "and",
                        [ "eq", "SUBSYSTEM", "usb-serial" ],
                        [ "regex", "DEVNAME",
                                [ "^ttyUSB", "^ttyACM" ]
                        ]
                ],
                [ "exec", "/sbin/hotplug-call", "tty" ],
                [ "if",
                        [ "isdir", "/etc/hotplug.d/%SUBSYSTEM%" ],
                        [ "exec", "/sbin/hotplug-call", "%SUBSYSTEM%" ]
                ]
        ]
```
which results in scripts in /etc/hotplug.d/tty being executed with add and remove events when there is a match on usb-serial subsystem for a ttyUSB* or ttyACM* device.

I suppose a similar kind of thing is needed for the usb lp subsystem (although maybe doesn't require the regex bit).

Basically on add we want to run /etc/init.d/p9100d restart so that it comes back up when the device is hotplugged.

**Comment by triebelf** [comment]
Hi, finally the weekend, found some time to look at this again.
I see now that the process isn't started when device boots, so I changed the start symlink to /etc/rc.d/S99p910nd
Now it's coming up during start, but I don't see log messages during boot.
Should this be changed in general?

**Comment by cshoredaniel** [comment]
@triebelf Might need a proper procd trigger -- see https://github.com/openwrt/packages/pull/7863 for an example.

**Comment by cshoredaniel** [comment]
Seems network applications can come up before there are any interfaces.   Either that START=xx is missing (haven't looked yet).

**Comment by cshoredaniel** [comment]
So I discovered that if the network changes p910nd needs to be restarted -- I guess socket does something (or goes away) and p910nd doesn't like that and doesn't recover when it happens.

**Comment by paucoma** [comment]
I too have had this "issue" appear after "some time" which in my case is related to a network change/or unavailability because I have the p910nd service attached to the **wan interface**.

It appears after a router reboot and wan port disconnected/unavailable to assign address

---
my solution was by adding the following to the end of my **/etc/init.d/p910nd** script:

```
service_triggers(){
        procd_add_interface_trigger "interface.*.up" wan "/etc/init.d/p910nd" restart
        logger -t p910nd "Restarting p910nd service due to up trigger of wan"
}
```

This now restarts the p910nd service when the **wan interface goes up**

---
### Investigation documented

- After the first time it happend, I setup some additional p910nd logging [as described here](https://openwrt.org/docs/guide-user/services/print_server/p910nd.server#p910nd_process_stdout_to_system_log). 

Here is the log output (filtered to relevant data): Network cable on WAN port unplugged and rebooted:

```
Wed Nov 11 11:09:41 2020 kern.notice kernel: [    0.000000] Linux version 4.14.180 (builder@buildhost) (gcc version 7.5.0 (OpenWrt GCC 7.5.0 r11063-85e04e9f46)) #0 Sat May 16 18:32:20 2020
Wed Nov 11 11:09:41 2020 kern.info kernel: [    0.000000] bootconsole [early0] enabled
Wed Nov 11 11:09:41 2020 kern.info kernel: [    0.000000] CPU0 revision is: 00019374 (MIPS 24Kc)
Wed Nov 11 11:09:41 2020 kern.info kernel: [    0.000000] MIPS: machine is TP-Link TL-WR710N v1
Wed Nov 11 11:09:41 2020 kern.info kernel: [    0.000000] SoC: Atheros AR9330 rev 1
.
..
...
..
.
Wed Nov 11 11:09:41 2020 kern.notice kernel: [    2.561061] random: fast init done
Wed Nov 11 11:09:41 2020 user.info kernel: [    2.929541] init: Console is alive
Wed Nov 11 11:09:41 2020 user.info kernel: [    2.931906] init: - watchdog -
Wed Nov 11 11:09:41 2020 user.info kernel: [    4.156855] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.281172] usbcore: registered new interface driver usbfs
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.285315] usbcore: registered new interface driver hub
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.290631] usbcore: registered new device driver usb
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.306238] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.319334] ci_hdrc ci_hdrc.0: EHCI Host Controller
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.322888] ci_hdrc ci_hdrc.0: new USB bus registered, assigned bus number 1
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.351059] ci_hdrc ci_hdrc.0: USB 2.0 started, EHCI 1.00
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.356185] hub 1-0:1.0: USB hub found
Wed Nov 11 11:09:41 2020 kern.info kernel: [    4.359217] hub 1-0:1.0: 1 port detected
Wed Nov 11 11:09:41 2020 user.info kernel: [    4.365003] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Wed Nov 11 11:09:41 2020 user.info kernel: [    4.373359] init: - preinit -
.
..
...
..
.
Wed Nov 11 11:09:41 2020 kern.info kernel: [    7.532210] eth0: link up (1000Mbps/Full duplex)
Wed Nov 11 11:09:41 2020 kern.info kernel: [   10.178493] eth0: link down
Wed Nov 11 11:09:41 2020 user.info kernel: [   11.244685] procd: - init -
Wed Nov 11 11:09:41 2020 user.info kernel: [   12.316393] kmodloader: loading kernel modules from /etc/modules.d/*
Wed Nov 11 11:09:41 2020 kern.info kernel: [   12.801398] usblp 1-1:1.0: usblp0: USB Bidirectional printer dev 2 if 0 alt 1 proto 2 vid 0x03F0 pid 0x0317
Wed Nov 11 11:09:41 2020 kern.info kernel: [   12.809981] usbcore: registered new interface driver usblp
Wed Nov 11 11:09:57 2020 kern.info kernel: [   32.901628] eth0: link up (1000Mbps/Full duplex)
Wed Nov 11 11:09:57 2020 kern.info kernel: [   32.911950] br-lan: port 1(eth0) entered blocking state
Wed Nov 11 11:09:57 2020 kern.info kernel: [   32.915731] br-lan: port 1(eth0) entered disabled state
Wed Nov 11 11:09:57 2020 kern.info kernel: [   32.921461] device eth0 entered promiscuous mode
Wed Nov 11 11:09:57 2020 kern.info kernel: [   32.963226] br-lan: port 1(eth0) entered blocking state
Wed Nov 11 11:09:57 2020 kern.info kernel: [   32.967008] br-lan: port 1(eth0) entered forwarding state
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'lan' is enabled
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'lan' is setting up now
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'lan' is now up
Wed Nov 11 11:09:57 2020 kern.info kernel: [   33.072060] br-wan: port 1(eth1) entered blocking state
Wed Nov 11 11:09:57 2020 kern.info kernel: [   33.075834] br-wan: port 1(eth1) entered disabled state
Wed Nov 11 11:09:57 2020 kern.info kernel: [   33.081582] device eth1 entered promiscuous mode
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'wan' is enabled
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'loopback' is enabled
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'loopback' is setting up now
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'loopback' is now up
Wed Nov 11 11:09:57 2020 daemon.notice netifd: bridge 'br-lan' link is up
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'lan' has link connectivity
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Network device 'eth0' link is up
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Network device 'lo' link is up
Wed Nov 11 11:09:57 2020 daemon.notice netifd: Interface 'loopback' has link connectivity
Wed Nov 11 11:10:00 2020 daemon.info p910nd[1379]: bind: Address not available
Wed Nov 11 11:10:00 2020 daemon.info p910nd[1379]: accept: Bad file descriptor
Wed Nov 11 11:10:01 2020 daemon.info procd: - init complete -
Wed Nov 11 11:10:03 2020 daemon.notice netifd: Interface 'wan' has link connectivity
Wed Nov 11 11:10:03 2020 daemon.notice netifd: Interface 'wan' is setting up now
Wed Nov 11 11:10:03 2020 daemon.notice netifd: wan (1462): udhcpc: started, v1.30.1
Wed Nov 11 11:10:03 2020 daemon.notice netifd: wan (1462): udhcpc: sending discover
Wed Nov 11 11:10:03 2020 daemon.notice netifd: wan (1462): udhcpc: performing DHCP renew
Wed Nov 11 11:10:03 2020 daemon.notice netifd: wan (1462): udhcpc: sending discover
Wed Nov 11 11:10:06 2020 daemon.info p910nd[1508]: bind: Address not available
Wed Nov 11 11:10:06 2020 daemon.info p910nd[1508]: accept: Bad file descriptor
Wed Nov 11 11:10:06 2020 daemon.notice netifd: wan (1462): udhcpc: sending discover
Wed Nov 11 11:10:11 2020 daemon.info p910nd[1513]: bind: Address not available
Wed Nov 11 11:10:11 2020 daemon.info p910nd[1513]: accept: Bad file descriptor
Wed Nov 11 11:10:16 2020 daemon.info p910nd[1514]: bind: Address not available
Wed Nov 11 11:10:16 2020 daemon.info p910nd[1514]: accept: Bad file descriptor
Wed Nov 11 11:10:21 2020 daemon.info p910nd[1515]: bind: Address not available
Wed Nov 11 11:10:21 2020 daemon.info p910nd[1515]: accept: Bad file descriptor
Wed Nov 11 11:10:26 2020 daemon.info p910nd[1516]: bind: Address not available
Wed Nov 11 11:10:26 2020 daemon.info p910nd[1516]: accept: Bad file descriptor
Wed Nov 11 11:10:26 2020 daemon.info procd: Instance p910nd::p9100d s in a crash loop 6 crashes, 0 seconds since last crash
Wed Nov 11 11:10:39 2020 kern.info kernel: [   75.293017] eth1: link up (100Mbps/Full duplex)
Wed Nov 11 11:10:39 2020 kern.info kernel: [   75.296167] br-wan: port 1(eth1) entered blocking state
Wed Nov 11 11:10:39 2020 kern.info kernel: [   75.301385] br-wan: port 1(eth1) entered forwarding state
Wed Nov 11 11:10:39 2020 daemon.notice netifd: Network device 'eth1' link is up
Wed Nov 11 11:10:40 2020 daemon.notice netifd: wan (1462): udhcpc: sending select for 192.168.11.156
Wed Nov 11 11:10:40 2020 daemon.notice netifd: wan (1462): udhcpc: lease of 192.168.11.156 obtained, lease time 43200
Wed Nov 11 11:10:40 2020 daemon.notice netifd: Interface 'wan' is now up
```

as can be seen : 

first attempt 
```
11:10:00 2020 daemon.info p910nd[1379]: bind: Address not available
11:10:00 2020 daemon.info p910nd[1379]: accept: Bad file descriptor
```
5 more attempts spaced 5 seconds appart and crash
```
11:10:26 2020 daemon.info procd: Instance p910nd::p9100d s in a crash loop 6 crashes, 0 seconds since last crash
```
connected wan cable 10 seconds later
```
11:10:39 2020 kern.info kernel: [   75.293017] eth1: link up (100Mbps/Full duplex)
11:10:39 2020 kern.info kernel: [   75.296167] br-wan: port 1(eth1) entered blocking state
11:10:39 2020 kern.info kernel: [   75.301385] br-wan: port 1(eth1) entered forwarding state
11:10:39 2020 daemon.notice netifd: Network device 'eth1' link is up
11:10:40 2020 daemon.notice netifd: wan (1462): udhcpc: sending select for 192.168.11.156
11:10:40 2020 daemon.notice netifd: wan (1462): udhcpc: lease of 192.168.11.156 obtained, lease time 43200
11:10:40 2020 daemon.notice netifd: Interface 'wan' is now up
```

The service had not started and was not in the active processes list. Printing of course didn't work.
A `/etc/init.d/p910nd restart` started sucessfully the print server and could print then no problem.

Next test, I then rebooted the router maintaining the wan cable connected.

```
Wed Nov 11 12:19:11 2020 daemon.notice netifd: Interface 'wan' is enabled
Wed Nov 11 12:19:13 2020 kern.info kernel: [   35.133054] eth1: link up (100Mbps/Full duplex)
Wed Nov 11 12:19:13 2020 kern.info kernel: [   35.136181] br-wan: port 1(eth1) entered blocking state
Wed Nov 11 12:19:13 2020 kern.info kernel: [   35.141408] br-wan: port 1(eth1) entered forwarding state
Wed Nov 11 12:19:13 2020 daemon.notice netifd: Network device 'eth1' link is up
Wed Nov 11 12:19:13 2020 daemon.notice netifd: bridge 'br-wan' link is up
Wed Nov 11 12:19:13 2020 daemon.notice netifd: Interface 'wan' has link connectivity
Wed Nov 11 12:19:13 2020 daemon.notice netifd: Interface 'wan' is setting up now
Wed Nov 11 12:19:14 2020 daemon.notice netifd: wan6 (1363): udhcpc: started, v1.30.1
Wed Nov 11 12:19:14 2020 daemon.notice netifd: wan (1367): udhcpc: started, v1.30.1
Wed Nov 11 12:19:15 2020 daemon.notice netifd: wan (1367): udhcpc: sending discover
Wed Nov 11 12:19:15 2020 daemon.notice netifd: wan (1367): udhcpc: sending select for 192.168.11.156
Wed Nov 11 12:19:15 2020 daemon.notice netifd: wan (1367): udhcpc: lease of 192.168.11.156 obtained, lease time 43200
Wed Nov 11 12:19:15 2020 daemon.info p910nd[1400]: bind: Address not available
Wed Nov 11 12:19:15 2020 daemon.info p910nd[1400]: accept: Bad file descriptor
Wed Nov 11 12:19:15 2020 daemon.notice netifd: wan6 (1363): udhcpc: sending discover
Wed Nov 11 12:19:16 2020 daemon.info procd: - init complete -
Wed Nov 11 12:19:17 2020 daemon.notice netifd: Interface 'wan' is now up
```

first attempt failed aswell
```
12:19:15 2020 daemon.info p910nd[1400]: bind: Address not available
12:19:15 2020 daemon.info p910nd[1400]: accept: Bad file descriptor
```
but a couple seconds later the wan interface came up and the second try probably succeeded and the service was working.

----

Solutions:

i noticed in the system log the following message:
```
user.notice firewall: Reloading firewall due to ifup of wan (br-wan)
```

procd trigger method:

So started researching and found:
- [old openwrt wiki on procd](https://oldwiki.archive.openwrt.org/inbox/procd-init-scripts)
- [openwrt wiki on procd](https://openwrt.org/docs/guide-developer/procd-init-scripts#specifying_triggers)
- foundout how to list the triggers of a specific interface ` ubus -v list network.interface.wan`

and came up with a solution by adding the following to the end of my **/etc/init.d/p910nd** script:

```
service_triggers(){
        procd_add_interface_trigger "interface.*.up" wan "/etc/init.d/p910nd" restart
        logger -t p910nd "Restarting p910nd service due to up trigger of wan"
}
```
This now restarts the p910nd service when the **wan interface goes up**

```
Wed Nov 11 15:45:50 2020 kern.notice kernel: [    0.000000] Linux version 4.14.180 (builder@buildhost) (gcc version 7.5.0 (OpenWrt GCC 7.5.0 r11063-85e04e9f46)) #0 Sat May 16 18:32:20 2020
Wed Nov 11 15:45:50 2020 user.info kernel: [    2.929349] init: Console is alive
Wed Nov 11 15:45:50 2020 user.info kernel: [    2.931722] init: - watchdog -
Wed Nov 11 15:45:50 2020 user.info kernel: [    4.157021] kmodloader: loading kernel modules from /etc/modules-boot.d/*
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.281349] usbcore: registered new interface driver usbfs
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.285484] usbcore: registered new interface driver hub
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.290803] usbcore: registered new device driver usb
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.306413] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.319520] ci_hdrc ci_hdrc.0: EHCI Host Controller
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.323078] ci_hdrc ci_hdrc.0: new USB bus registered, assigned bus number 1
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.351059] ci_hdrc ci_hdrc.0: USB 2.0 started, EHCI 1.00
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.356184] hub 1-0:1.0: USB hub found
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.359214] hub 1-0:1.0: 1 port detected
Wed Nov 11 15:45:50 2020 user.info kernel: [    4.365004] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
Wed Nov 11 15:45:50 2020 user.info kernel: [    4.373362] init: - preinit -
Wed Nov 11 15:45:50 2020 kern.info kernel: [    4.965118] usb 1-1: new full-speed USB device number 2 using ci_hdrc
Wed Nov 11 15:45:50 2020 kern.info kernel: [    7.532210] eth0: link up (1000Mbps/Full duplex)
Wed Nov 11 15:45:50 2020 kern.info kernel: [   10.178409] eth0: link down
Wed Nov 11 15:45:50 2020 user.info kernel: [   10.198062] procd: - early -
Wed Nov 11 15:45:50 2020 user.info kernel: [   10.199639] procd: - watchdog -
Wed Nov 11 15:45:50 2020 user.info kernel: [   10.995789] procd: - watchdog -
Wed Nov 11 15:45:50 2020 user.info kernel: [   10.997927] procd: - ubus -
Wed Nov 11 15:45:50 2020 user.info kernel: [   11.244093] procd: - init -
Wed Nov 11 15:45:50 2020 user.info kernel: [   12.333002] kmodloader: loading kernel modules from /etc/modules.d/*
Wed Nov 11 15:45:50 2020 kern.info kernel: [   12.809465] usblp 1-1:1.0: usblp0: USB Bidirectional printer dev 2 if 0 alt 1 proto 2 vid 0x03F0 pid 0x0317
Wed Nov 11 15:45:50 2020 kern.info kernel: [   12.818101] usbcore: registered new interface driver usblp
Wed Nov 11 15:45:50 2020 user.info kernel: [   13.531673] kmodloader: done loading kernel modules from /etc/modules.d/*
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.023504] eth0: link up (1000Mbps/Full duplex)
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.027457] br-lan: port 1(eth0) entered blocking state
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.031986] br-lan: port 1(eth0) entered disabled state
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.037518] device eth0 entered promiscuous mode
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.079627] br-lan: port 1(eth0) entered blocking state
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.083481] br-lan: port 1(eth0) entered forwarding state
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'lan' is enabled
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'lan' is setting up now
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'lan' is now up
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.206081] br-wan: port 1(eth1) entered blocking state
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.209865] br-wan: port 1(eth1) entered disabled state
Wed Nov 11 15:46:06 2020 kern.info kernel: [   33.215625] device eth1 entered promiscuous mode
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'wan' is enabled
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'loopback' is enabled
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'loopback' is setting up now
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'loopback' is now up
Wed Nov 11 15:46:06 2020 daemon.notice netifd: bridge 'br-lan' link is up
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'lan' has link connectivity
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Network device 'eth0' link is up
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Network device 'lo' link is up
Wed Nov 11 15:46:06 2020 daemon.notice netifd: Interface 'loopback' has link connectivity
Wed Nov 11 15:46:07 2020 user.notice firewall: Reloading firewall due to ifup of lan (br-lan)
Wed Nov 11 15:46:09 2020 user.notice p910nd: Restarting p910nd service due to up trigger of wan
Wed Nov 11 15:46:10 2020 daemon.info p910nd[1380]: bind: Address not available
Wed Nov 11 15:46:10 2020 daemon.info p910nd[1380]: accept: Bad file descriptor
Wed Nov 11 15:46:11 2020 daemon.info procd: - init complete -
Wed Nov 11 15:46:12 2020 daemon.notice netifd: bridge 'br-wan' link is up
Wed Nov 11 15:46:12 2020 daemon.notice netifd: Interface 'wan' has link connectivity
Wed Nov 11 15:46:12 2020 daemon.notice netifd: Interface 'wan' is setting up now
Wed Nov 11 15:46:12 2020 daemon.notice netifd: wan (1463): udhcpc: started, v1.30.1
Wed Nov 11 15:46:12 2020 daemon.notice netifd: wan (1463): udhcpc: sending discover
Wed Nov 11 15:46:12 2020 daemon.notice netifd: wan (1463): udhcpc: performing DHCP renew
Wed Nov 11 15:46:12 2020 daemon.notice netifd: wan (1463): udhcpc: sending discover
Wed Nov 11 15:46:15 2020 daemon.info p910nd[1513]: bind: Address not available
Wed Nov 11 15:46:15 2020 daemon.info p910nd[1513]: accept: Bad file descriptor
Wed Nov 11 15:46:15 2020 daemon.notice netifd: wan (1463): udhcpc: sending discover
Wed Nov 11 15:46:20 2020 daemon.info p910nd[1514]: bind: Address not available
Wed Nov 11 15:46:20 2020 daemon.info p910nd[1514]: accept: Bad file descriptor
Wed Nov 11 15:46:25 2020 daemon.info p910nd[1515]: bind: Address not available
Wed Nov 11 15:46:25 2020 daemon.info p910nd[1515]: accept: Bad file descriptor
Wed Nov 11 15:46:30 2020 daemon.info p910nd[1516]: bind: Address not available
Wed Nov 11 15:46:30 2020 daemon.info p910nd[1516]: accept: Bad file descriptor
Wed Nov 11 15:46:35 2020 daemon.info p910nd[1517]: bind: Address not available
Wed Nov 11 15:46:35 2020 daemon.info p910nd[1517]: accept: Bad file descriptor
Wed Nov 11 15:46:35 2020 daemon.info procd: Instance p910nd::p9100d s in a crash loop 6 crashes, 0 seconds since last crash
Wed Nov 11 15:47:01 2020 kern.info kernel: [   87.772966] eth1: link up (100Mbps/Full duplex)
Wed Nov 11 15:47:01 2020 kern.info kernel: [   87.776094] br-wan: port 1(eth1) entered blocking state
Wed Nov 11 15:47:01 2020 kern.info kernel: [   87.781339] br-wan: port 1(eth1) entered forwarding state
Wed Nov 11 15:47:01 2020 daemon.notice netifd: Network device 'eth1' link is up
Wed Nov 11 15:47:03 2020 daemon.notice netifd: wan (1463): udhcpc: sending select for 192.168.11.156
Wed Nov 11 15:47:04 2020 daemon.notice netifd: wan (1463): udhcpc: lease of 192.168.11.156 obtained, lease time 43200
Wed Nov 11 15:47:04 2020 daemon.notice netifd: Interface 'wan' is now up
Wed Nov 11 15:47:04 2020 user.notice firewall: Reloading firewall due to ifup of wan (br-wan)
Wed Nov 11 15:47:05 2020 user.notice p910nd: Restarting p910nd service due to up trigger of wan
```

I can also see the registered triggers and configuration applied with: ` ubus call service list '{"name":"p910nd","verbose": true}'` 

```
root@OpenWrt:~# ubus call service list '{"name":"p910nd","verbose": true}'
{
        "p910nd": {
                "instances": {
                        "p9100d": {
                                "running": true,
                                "pid": 1588,
                                "command": [
                                        "/usr/sbin/p910nd",
                                        "-d",
                                        "-b",
                                        "-f",
                                        "/dev/usb/lp0",
                                        "-i",
                                        "192.168.11.156",
                                        "0"
                                ],
                                "term_timeout": 5,
                                "respawn": {
                                        "threshold": 3600,
                                        "timeout": 5,
                                        "retry": 5
                                },
                                "user": "p910nd"
                        }
                },
                "triggers": [
                        [
                                "interface.*.up",
                                [
                                        "if",
                                        [
                                                "eq",
                                                "interface",
                                                "wan"
                                        ],
                                        [
                                                "run_script",
                                                "/etc/init.d/p910nd",
                                                "restart"
                                        ]
                                ],
                                1000
                        ]
                ]
        }
}

```

you can monitor when the trigger is invoked with: 
```
root@OpenWrt:~# ubus monitor | grep '"method":"up","data":{"interface":"wan"}}'
<- 38326d9b #14a31b7d         invoke: {"objid":346233725,"method":"up","data":{"interface":"wan"}}
```

**Comment by pprindeville** [comment]
Why would a printer inside the perimeter need `wan` to be up?  I don't get it.

**Comment by paucoma** [comment]
> Why would a printer inside the perimeter need `wan` to be up? I don't get it.

In my case its a subnetwork, I don't want the LAN or the Wifi created and managed by the router to have access to whats connected to the WAN but I would like to print to the printer which is connected to the usb port of the router from sources connected to the WAN.

**Comment by ptpt52** [comment]
the bind option in openwrt config is the ipaddr, which is determined during configuration
This may not be correct, because the interface may not have an ip address at that time.

**Comment by GeorgeSapkin** [comment]
Closing as stale.
